src/grokcore/component/util.py

```diff
@@ -128,7 +128,9 @@ def provideInterface(id, interface, iface_type=None, info=''):
         id = "%s.%s" % (interface.__module__, interface.__name__)
 
     if not IInterface.providedBy(interface):
-        if not isinstance(interface, (type, types.ClassType)):
+        # python3 compatible
+        if not isinstance(interface, type) or (hasattr(types, 'ClassType') and
+                                               isinstance(interface, types.ClassType)):
```

Comment:
I recommend [the six approach](https://pythonhosted.org/six/index.html?highlight=class_types#six.class_types): in compat.py define `class_types = (type,)` or `class_types = (type, types.ClassType)` once, then import it here and do `if not isinstance(type, class_types)` here.